### PR TITLE
Flowers OOP with raylib/examples/models/resources/skybox.png

### PR DESCRIPTION
**Pull request by coderextreme**
Example with skybox and object oriented programming.  Also intended to show reflection and prismatic effects (chromatic dispersion) with shaders, but I'm really unsure about if my skybox works right.   The skybox in this pull request is pulled from raylib/examples/models/resources/skybox.png.  I drew inspiration from the skybox example in raylib/examples, but it's not a complete port.

A good skybox image should show reflection, refraction and chromatic dispersion.  Here's one with a skybox from Paul Debevec:  
image

### CODE REVIEW

**Comment by coderextreme** [comment]
I got different results by calling reset.  I'm not sure what's up.

**Comment by coderextreme** [comment]
Also, does not compile to JS:

$ nim js flowers_oop.nim
Hint: used config file 'C:\Users\jcarl\Downloads\nim-2.2.0\config\nim.cfg' [Conf]
Hint: used config file 'C:\Users\jcarl\Downloads\nim-2.2.0\config\config.nims' [Conf]
Hint: used config file 'C:\Users\jcarl\raylib-examples\config.nims' [Conf]
...........................................................................
C:\Users\jcarl\.nimble\pkgs2\naylib-24.44.1-3eb3b966739400237ccd37c0bc84b410f21496f4\raylib.nim(1708, 1) Error: signature for '=destroy' must be proc[T: object](x: var T)


I am not sure what's up, I'll try to upgrade naylib

File: models/flowers_oop.nim [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+    build(flowers[f])
+
+  defer:
+    unloadTexture(backgroundTexture.id)
+    unloadTexture(cubemap.id)
+    # unloadModel(skybox.id)
+    # unloadShader(skyboxShader.id)
+    # unloadImage(img.id)
+    for f in 0..<flowers.len:
+      `=destroy`(flowers[f])
```
Comment (planetis-m) [inline models/flowers_oop.nim]: Destructors are run automatically. No need to call them explicitly. If you really need to (such as to free an image after loading it into a texture) and you don't depend on scopes, there's `reset`.

File: models/flowers_oop.nim [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+  #if dest.modelShader.id != 0:
+  #  unloadShader(dest.modelShader.id)
+  #if dest.model.meshCount != 0:
+  #  unloadModel(dest.model.id)
+
+  dest = source
+  # Zero out source to prevent double-free
+  #source.cubemap.id = 0
+  #source.modelShader.id = 0
+  #source.model.meshCount = 0
```
Comment (planetis-m) [inline models/flowers_oop.nim]: All these destructor hooks should be removed. The raylib library handles destructors for you. (If you need to attach a destructor to an object that has other managed resources you call them explicitly on the fields).

**Review by coderextreme — COMMENTED** [review]
Removed explicit destructors except those on Flower instances (deferred)

**Comment by coderextreme** [comment]
Ready to review again.  JS transpiling still doesn't work.   I don't know how to solve it,  I don't think claude.ai does eitther, at least not in my code.

**Comment by planetis-m** [comment]
> Ready to review again. JS transpiling still doesn't work. I don't know how to solve it, I don't think claude.ai does eitther, at least not in my code.

Naylib doesn't support the js platform since raylib is written in C. But you can compile it with emscripten. I have sent you a link on discord.

File: models/flowers_oop.nim [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+  defer: closeWindow()
+
+  var cube = genMeshCube(4.0f, 4.0f, 4.0f)
+  var skybox = loadModelFromMesh(cube)
+  var skyboxShader = loadShaderFromMemory(vertexSkyboxShader, fragmentSkyboxShader)
+  skybox.materials[0].shader = skyboxShader
+
+  var image = loadImage("resources/skybox.png")  # TODO Replace with your texture path
+  if image.data == nil:
+    raise newException(IOError, "Failed to load skybox.png texture")
```
Comment (planetis-m) [inline models/flowers_oop.nim:409]: Btw these functions have error checking, so you don't need to do that explicitly. Apparently only loadShaderFromMemory is missing a isShaderValid call. Will investigate why.

**Comment by planetis-m** [comment]
Merging anyway. Thank you!

File: models/flowers_oop.nim (hunk 3 above)
Comment (coderextreme) [reply models/flowers_oop.nim:409]: I wouldn't put the checking in myself (I'm kind of bad that way), so I'm glad the library handles it!

Comment (coderextreme) [reply models/flowers_oop.nim:409]: I wouldn't put the checking in myself (I'm kind of bad that way), so I'm glad the library handles it!

**Comment by coderextreme** [comment]
> > Ready to review again. JS transpiling still doesn't work. I don't know how to solve it, I don't think claude.ai does eitther, at least not in my code.
> 
> 
> 
> Naylib doesn't support the js platform since raylib is written in C. But you can compile it with emscripten. I have sent you a link on discord.

What I did was use claude.ai to translate the code to Three.js.  It got too aggressive with the normal in my shader though.  See: https://coderextreme.net/assetmanager/ The translation worked well, and even helped fix the skybox issues.  I just had to specify 6 images instead of 1.  I don't know if naylib supports 6 images or not.

**Pull request merged**